.pre-commit-config.yaml

```
@@ -15,22 +10,12 @@ repos:
           - id: check-yaml
           - id: debug-statements
             language_version: python3
-    - repo: https://github.com/PyCQA/flake8
-      rev: 7.0.0
-      hooks:
-          - id: flake8
-            language_version: python3
-            additional_dependencies: [flake8-typing-imports>=1.9.0]
-    - repo: https://github.com/asottile/reorder_python_imports
-      rev: v3.12.0
-      hooks:
-          - id: reorder-python-imports
-            args: ['--application-directories=.:src', --py3-plus]
-    - repo: https://github.com/asottile/pyupgrade
-      rev: v3.15.1
+    - repo: https://github.com/astral-sh/ruff-pre-commit
+      rev: v0.2.2
       hooks:
-          - id: pyupgrade
-            args: [--py38-plus]
+        - id: ruff-format
+          args: ["--preview"]
+        - id: ruff
```

Let's follow the lead of https://github.com/sphinx-contrib/sphinx-lint/pull/110 and add:

```suggestion
        - id: ruff
          args: ["--exit-non-zero-on-fix"]
```